lib/utils/resolveObject.js

```diff
@@ -62,36 +66,29 @@ function traverseObj(obj) {
         traverseObj( obj[key] );
       } else {
         if (typeof obj[key] === 'string' && obj[key].indexOf('{') > -1) {
-          try {
-            obj[key] = compile_value( obj[key], [key] );
-          }
-          catch (e) {
-            errors.push(e);
-          }
+          obj[key] = compile_value( obj[key], [current_context.join('.')] );
         }
       }
       current_context.pop();
     }
   }
 
-  if(errors.length) {
-    throw new Error(errors.length===1 ? errors[0] : 'Failed due to '+errors.length+' errors:\n' + errors.join('\n').replace(/Error: /g,'\t'));
-  }
-
   return obj;
 }
 
 
-
+var foundCirc = {};
 function compile_value(value, stack) {
+
   var to_ret, ref;
-  stack.push(value.slice(1, -1));
 
   // Replace the reference inline, but don't replace the whole string because
   // references can be part of the value such as "1px solid {color.border.light}"
   value.replace(regex, function(match, variable) {
+    stack.push(variable);
+
     // Find what the value is referencing
-    ref = selfRef(variable.trim(), updated_object);
+    ref = selfRef(variable.trim(), updated_object, stack);
```

Comment:
Why are we passing stack here, the method does not seem to use this argument.

Reply:
Yeah... I was thinking about removing current_context as a private external variable so that it was passed in using stack instead. What do you think is right here?